Fix average consumption calculation

lomamech opened the pull request:
- Improvement of the way to obtain the number of months necessary to
  calculate the average monthly consumption
- Implement integration test for Automaticallu calculated key values

lomamech committed:
bug(Average Consumption Calculation)
- Improvement of the way to obtain the number of months necessary to
  calculate the average monthly consumption

closes #4246

lomamech committed:
test(Stock Inventories Depots)
- Implement integration test for Automaticallu calculated key values

mbayopanda commented on server/controllers/stock/index.js:
```diff
@@ -382,9 +382,9 @@ function listLotsDepot(req, res, next) {
 function listInventoryDepot(req, res, next) {
   const params = req.query;
-  params.monthAverageConsumption = req.session.enterprise.settings.month_average_consumption;
+  const monthAverageConsumption = req.session.enterprise.settings.month_average_consumption;
```
👍

mbayopanda reviewed (commented):
@lomamech la fonction `processStockConsumptionAverage` est aussi utilisée dans la fonction `getLotsDepot`

jniles commented on test/integration/stock/stock.js:
```diff
@@ -183,4 +183,31 @@ describe('(/stock/) The Stock HTTP API', () => {
   );
+
+  it(
+    `/stock/inventories/depots Get Inventories in Stock By Depot`,
```
👍 🎆

lomamech committed:
Bug: add param monthAverageConsumption in Params of function listLotsDepot

lomamech committed:
bug(Report Inventories Report)
- Addition of the parameter monthAverageConsumption so that the [Stock]
  Inventories report can calculate
the average monthly consumption with the value defined by enterprise

lomamech committed:
Test: simplify integration test or Stock Inventories in depot

mbayopanda reviewed (approved):
LGTM 👍 

bors r+
